core/src/main/java/de/mirkosertic/bytecoder/core/backend/wasm/WasmStructuredControlflowCodeGenerator.java

```diff
@@ -1128,6 +1358,23 @@ private WasmValue toWasmValue(final InvokeDynamicExpression value) {
             } else {
                 throw new IllegalArgumentException("Not supported method " + bootstrapMethod.methodName + " on " + bootstrapMethod.className);
             }
+        } else if (bootstrapMethod.className.getClassName().equals("java.lang.runtime.ObjectMethods")) {
+            if ("bootstrap".equals(bootstrapMethod.methodName)) {
+                final ObjectString operation = (ObjectString) resolveCallsite.incomingDataFlows[1];
+                final String operationStr = compileUnit.getConstantPool().getPooledStrings().get(operation.value.index);
+                if ("toString".equals(operationStr)) {
+                    return generateInvokeDynamicObjectMethodsToString(value, resolveCallsite);
+                } else if ("hashCode".equals(operationStr)) {
+                    return ConstExpressions.i32.c(0);
+                    //return generateInvokeDynamicObjectMethodsHashCode(value, resolveCallsite);
```

Comment:
Why is this commented out?

Reply:
Wasm and JS backend currently do not provide good hashCodes for Record classes. In fact, their hash code is always 0. This is by far not optimal, but complies to the equals/hashCode contract of the JVM. Improving this is not in scope of this bug report from my point of view.